apps/core/src/contexts/HiddenAssetsProvider.tsx

```diff
@@ -0,0 +1,143 @@
+    showAsset: (assetId: string) => Promise<string | undefined>;
+    undoShowAsset: (assetId: string) => Promise<void>;
+}
+
+export const HiddenAssetsContext = createContext<HiddenAssetContext>({
+    hiddenAssets: {
+        type: 'loading',
+    },
+    setHiddenAssetIds: () => {},
+    hideAsset: async () => undefined,
+    undoHideAsset: async () => undefined,
+    showAsset: async () => undefined,
+    undoShowAsset: async () => undefined,
+});
+
+export const HiddenAssetsProvider = ({ children }: { children: ReactNode }) => {
+    const [hiddenAssets, setHiddenAssets] = useState<HiddenAssets>({
+        type: 'loading',
+    });
+
+    const hiddenAssetIds = hiddenAssets.type === 'loaded' ? hiddenAssets.assetIds : [];
+
+    useEffect(() => {
+        (async () => {
+            const hiddenAssets = (await get<string[]>(HIDDEN_ASSET_IDS)) ?? [];
+            setHiddenAssetIds(hiddenAssets);
+        })();
+    }, []);
+
+    function setHiddenAssetIds(hiddenAssetIds: string[]) {
+        setHiddenAssets({
+            type: 'loaded',
+            assetIds: hiddenAssetIds,
+        });
+    }
+
+    const hideAssetId = useCallback(
+        async (newAssetId: string) => {
+            if (hiddenAssetIds.includes(newAssetId)) return;
+
+            const newHiddenAssetIds = [...hiddenAssetIds, newAssetId];
+            setHiddenAssetIds(newHiddenAssetIds);
+            await set(HIDDEN_ASSET_IDS, newHiddenAssetIds);
+            return newAssetId;
+        },
+        [hiddenAssetIds],
+    );
+
+    const undoHideAsset = async (assetId: string) => {
+        try {
+            let updatedHiddenAssetIds;
+            setHiddenAssets((previous) => {
+                const previousIds = previous.type === 'loaded' ? previous.assetIds : [];
+                updatedHiddenAssetIds = previousIds.filter((id) => id !== assetId);
+                return {
+                    type: 'loaded',
+                    assetIds: updatedHiddenAssetIds,
+                };
+            });
+            await set(HIDDEN_ASSET_IDS, updatedHiddenAssetIds);
+        } catch (error) {
+            // Restore the asset ID back to the hidden asset IDs list
+            setHiddenAssetIds([...hiddenAssetIds, assetId]);
+            await set(HIDDEN_ASSET_IDS, hiddenAssetIds);
+        }
+    };
+
+    const showAssetId = useCallback(
+        async (newAssetId: string) => {
+            if (!hiddenAssetIds.includes(newAssetId)) return;
+
+            try {
+                const updatedHiddenAssetIds = hiddenAssetIds.filter((id) => id !== newAssetId);
+                setHiddenAssetIds(updatedHiddenAssetIds);
+                await set(HIDDEN_ASSET_IDS, updatedHiddenAssetIds);
+                return newAssetId;
+            } catch (error) {
+                // Restore the asset ID back to the hidden asset IDs list
+                setHiddenAssetIds([...hiddenAssetIds, newAssetId]);
+                await set(HIDDEN_ASSET_IDS, hiddenAssetIds);
+            }
```

Comment:
1. You can just set the state after calling `set`
2. If the first call to `set` can throw an error, why would the second one (in `catch`) be successful?
```suggestion
			const updatedHiddenAssetIds = hiddenAssetIds.filter((id) => id !== newAssetId);
            try {
                await set(HIDDEN_ASSET_IDS, updatedHiddenAssetIds);
            	setHiddenAssetIds(updatedHiddenAssetIds);
            } catch (_) {
                // Keep the asset hidden
            }
            return newAssetId;
```